src/shogun/classifier/Perceptron.cpp

```diff
@@ -73,9 +74,11 @@ void CPerceptron::iteration()
 		{
 			converged = false;
 			const auto gradient = learn_rate * true_label;
-			put("bias", bias + gradient);
+			bias += gradient;
 			v.add(gradient, w);
-			put("w", w);
+
+			observe<SGVector<float64_t>>(m_current_iteration, "w");
+			observe<float64_t>(m_current_iteration, "bias");
```

Comment:
question. If the step is passed here, then we don't need it inside SGObject anymore right? So that problem is also solved?

Reply:
Yes, the problem is solved since we will need to specify the `step` param directly from now on (and we won't need to infer it).